buildSrc/src/main/java/org/elasticsearch/gradle/EnforceDeprecationFailuresPlugin.java

```diff
@@ -78,15 +78,13 @@ private void failOnCompileConfigurationResolution(SourceSet sourceSet) {
             .getByName(sourceSet.getCompileConfigurationName())
             .getIncoming()
             .beforeResolve(resolvableDependencies -> {
-                if (resolvableDependencies.getDependencies().size() > 0) {
```

Comment:
We should always fail on resolution. no matter if the configuration actually contains a dependency or not.